cmd/haplotypeGenerator/haplotypeGenerator.go

```diff
@@ -6,83 +6,83 @@ package main
 import (
 	"flag"
 	"fmt"
-	"log"
-
 	"github.com/vertgenlab/gonomics/bed"
 	"github.com/vertgenlab/gonomics/dna"
 	"github.com/vertgenlab/gonomics/exception"
 	"github.com/vertgenlab/gonomics/fasta"
 	"github.com/vertgenlab/gonomics/fileio"
 	"github.com/vertgenlab/gonomics/interval"
 	"github.com/vertgenlab/gonomics/vcf"
+	"log"
 )
 
-func haplotypeGenerator(genomeFile string, snpFile string, regionFile string, outdir string) {
-	var currState int
-	variants, header := vcf.Read(snpFile)
-	variantIntervals := make([]interval.Interval, len(variants))
-
-	// fmt.Println("started ok")
-
-	for i := range variants {
-		variantIntervals[i] = variants[i]
-	}
-
-	tree := interval.BuildTree(variantIntervals)
+type Settings struct {
+	ReferenceGenomeFile string
+	VcfFile             string
+	RegionBedFile       string
```

Comment:
What is the purpose of the bed file? To subset the genome into regions that you care about? Concievably, this would be a no-gap.bed in most cases?

Reply:
or is the bed file telling you were haplotype blocks exist?